go/master/service.go

```diff
@@ -77,11 +77,13 @@ type taskEntry struct {
 	NumFailure int
 }
 
-type taskQueues struct {
-	Todo    []taskEntry
-	Pending map[int]taskEntry // map from task ID to task entry
-	Done    []taskEntry
-	Failed  []taskEntry
+type masterState struct {
+	Todo     []taskEntry
+	Pending  map[int]taskEntry // map from task ID to task entry
+	Done     []taskEntry
+	Failed   []taskEntry
+	CurPass  int
+	JobTasks []taskEntry
```

Comment:
I think if a master gets restarted, it has no chance to fill in `JobTasks`, and later a pass finishes, todo queue will be set to `JobTasks` which is `nil`.